db/disk/course.go

```diff
@@ -24,16 +28,18 @@ func (this *backend) ClearCourse(course *model.Course) error {
 }
 
 func (this *backend) LoadCourse(path string) (*model.Course, error) {
-    this.lock.Lock();
-    defer this.lock.Unlock();
+    courseDir := util.ShouldAbs(path);
```

Comment:
This is a path (to a file) not a dir.

Note that "path" can be used for both a file and a dir (and technically many other things).
But, in this case the path should be pointing to a file (usually called "course.json" in this case).